sql-plugin/src/main/scala/com/nvidia/spark/rapids/AstUtil.scala

```diff
@@ -119,4 +122,112 @@ object AstUtil {
       }
     }
   }
+
+  abstract class NonAstJoinCondSplitStrategy() {
+    def remainedCond(): Option[Expression]
+
+    def processBuildSide(input: ColumnarBatch): ColumnarBatch = { input }
+
+    def processStreamSide(input: ColumnarBatch): ColumnarBatch = { input }
+
+    def processPostJoin(iter: Iterator[ColumnarBatch]): Iterator[ColumnarBatch] = { iter }
```

Comment:
Similarly wonder if these method names should use the word "project" rather than "process" since that's what they're doing.

Reply:
Hmm, this is to align the API for ```JoinCondSplitAsProject``` and ```JoinCondSplitAsPostFilter```.  Generally, they're both handling process after join happens.